ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -344,49 +356,55 @@ AmbryReplica getReplicaForPartitionOnNode(String hostname, int port, String part
      */
     @Override
     public void onLiveInstanceChange(List<LiveInstance> liveInstances, NotificationContext changeContext) {
-      logger.info("Live instance change triggered with: " + liveInstances);
-      Set<String> liveInstancesSet = new HashSet<>();
-      for (LiveInstance liveInstance : liveInstances) {
-        liveInstancesSet.add(liveInstance.getInstanceName());
-      }
-      for (String instanceName : allInstances) {
-        if (liveInstancesSet.contains(instanceName)) {
-          instanceNameToDataNodeIdMap.get(instanceName).setState(HardwareState.AVAILABLE);
-        } else {
-          instanceNameToDataNodeIdMap.get(instanceName).setState(HardwareState.UNAVAILABLE);
+      synchronized (notificationLock) {
+        logger.info("Live instance change triggered with: " + liveInstances);
```

Comment:
Wanted this as INFO only for the transitioning phase for studying the patterns. TRACE/DEBUG comes with excessive logging that is unnecessary unless chasing an issue. That said, I have changed this to TRACE as this consists of all instances, not just the changed ones (as I had thought originally).